sql/core/src/main/scala/org/apache/spark/sql/UDFRegistration.scala

```diff
@@ -115,6 +115,22 @@ class UDFRegistration private[sql] (functionRegistry: FunctionRegistry) extends
           def builder(e: Seq[Expression]) = ScalaUDF(func, dataType, e, inputTypes.getOrElse(Nil), Some(name), nullable)
           functionRegistry.registerFunction(name, builder)
           UserDefinedFunction(func, dataType, inputTypes).withName(name).withNullability(nullable)
```

Comment:
We can't directly call `register(name, func, deterministic = true, distinctLike = false)` here?